arcgis-android-toolkit/src/main/java/com/esri/arcgisruntime/toolkit/ar/ArcGISArView.kt

```diff
@@ -391,40 +575,53 @@ final class ArcGISArView : FrameLayout, DefaultLifecycleObserver, Scene.OnUpdate
      * @since 100.6.0
      */
     override fun onUpdate(frameTime: FrameTime?) {
-        arSceneView.arFrame?.camera?.let { arCamera ->
-            if (arCamera.trackingState == TrackingState.TRACKING) {
-                // create a Pair from the rotation quaternion and translation to create a TransformationMatrix
-                Pair(
-                    arCamera.displayOrientedPose.rotationQuaternion.map { it.toDouble() }.toDoubleArray(),
-                    arCamera.displayOrientedPose.translation.map { it.toDouble() }.toDoubleArray()
-                ).let {
+        arSceneView?.arFrame?.camera?.let { arCamera ->
+            isTracking = (arCamera.trackingState == TrackingState.TRACKING).or(locationDataSource != null)
```

Comment:
do you need to recalculate `isTracking` here?  Or should that be handled in the `startTracking`/`stopTracking` code?